src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,35 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_SIZE_LOGIN = 6;
+    private static final int MIN_SIZE_PASSWORD = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationUserException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_SIZE_LOGIN) {
+            throw new RegistrationUserException("Login must be at least " + MIN_SIZE_LOGIN + ": "
+                    + MIN_SIZE_LOGIN + " characters");
```

Comment:
The message for the exception should be 'Login must be at least 6 characters' instead of concatenating the same value twice. Just use 'Login must be at least ' + MIN_SIZE_LOGIN + ' characters'.